staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -72,12 +72,16 @@ type EgressSelection struct {
 // Connection provides the configuration for a single egress selection client.
 type Connection struct {
 	// Type is the type of connection used to connect from client to konnectivity server.
-	// Currently supported values are "http-connect" and "direct".
+	// Currently supported values are "http-connect", "http-connect-uds", "grpc-uds" and "direct".
```

Comment:
[http-connect, grpc] and [uds, tcp] are two layers. We should make them two fields, rather than using one field to represent all 4 permutations.

Maybe call one field "protocol", and the other one "transport"?